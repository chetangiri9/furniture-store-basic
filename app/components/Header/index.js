/**
 *
 * Header
 *
 */

import React from 'react';
import { Layout } from 'antd';
import styled from 'styled-components';
import { injectIntl } from 'react-intl';
import { colors } from '@themes';

const StyledHeader = styled(Layout.Header)`
  && {
    &.ant-layout-header {
      padding: 0 3rem;
      height: 7rem;
    }
    display: flex;
    align-items: center;
    justify-content: space-between;
    background-color: ${colors.primary};
  }
`;
const Title = styled.p`
  font-size: x-large;
  color: white;
`;

function Header(props) {
  return (
    <StyledHeader {...props} data-testid="header">
      <Title data-testid="title">We Sell Recliners</Title>
    </StyledHeader>
  );
}

export default injectIntl(Header);
