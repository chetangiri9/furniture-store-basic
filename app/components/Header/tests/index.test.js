/**
 *
 * Tests for Header
 *
 */

import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import Header from '../index';

describe('<Header />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<Header />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain title', () => {
    const { getAllByTestId } = renderWithIntl(<Header />);
    expect(getAllByTestId('title').length).toBe(1);
  });
});
