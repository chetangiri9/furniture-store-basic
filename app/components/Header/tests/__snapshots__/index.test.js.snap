// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render and match the snapshot 1`] = `
<body>
  <div>
    <header
      class="ant-layout-header Header__StyledHeader-wp2jxc-0 iNLvgP"
      data-testid="header"
      intl="[object Object]"
    >
      <p
        class="Header__Title-wp2jxc-1 cqlBDj"
        data-testid="title"
      >
        We Sell Recliners
      </p>
    </header>
  </div>
</body>
`;
